### ISSUE TITLE
"WARNING: the gcp auth plugin is deprecated" come up even if followed all the instructions

### ISSUE THREAD

**Issue by artemdumanov** [kubernetes/client-go#1092]
I'm working with the gke cluster and have got a warning for any kubectl command.

```
W0509 17:55:55.415853  113282 gcp.go:120] WARNING: the gcp auth plugin is deprecated in v1.22+, unavailable in v1.25+; use gcloud instead.
To learn more, consult https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke
```

Versions.

```
$ k version
W0509 18:01:34.450905  114335 gcp.go:120] WARNING: the gcp auth plugin is deprecated in v1.22+, unavailable in v1.25+; use gcloud instead.
To learn more, consult https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.0", GitCommit:"4ce5a8954017644c5420bae81d72b09b735c21f0", GitTreeState:"clean", BuildDate:"2022-05-03T13:46:05Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.15-gke.3400", GitCommit:"750002971a60d8a06e0a403c52724257f0f68481", GitTreeState:"clean", BuildDate:"2022-03-08T09:33:43Z", GoVersion:"go1.15.15b5", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.24) and server (1.20) exceeds the supported minor version skew of +/-1
```

I went to the [page with instructions](https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke) and installed a component with the `apt-get` command, updated bash, reconnect to the cluster and etc. But I still had been seeing the warning. 

I said ok, my bad. Lets do it right way and install the component with `gcloud components install gke-gcloud-auth-plugin` command. But I was not able becuase I had installed google cli with the apt, so I reinstalled the gcloud cli, and did all the required operations described in the instruction. The result is the same and lame warning that pretty annoyes me.

The bottom line is that I did follow every instruction described in the doc and 2 ways but still see the warning. There are 2 ways. Or I do something wrong or I take the audacity to say we got a bug here.

**Comment by ronisec** [comment]
This happened to me as well when upgrading kubectl to v1.24. I ended up downgrading back to v1.22

**Comment by adam-epam** [comment]
Can we have a way to disable this warning? It seems counter-productive.
Thank you!

**Comment by artemdumanov** [comment]
@adam-epam  As far as I've seen on the internet there is no way to suppress the warning in kubectl ;/
* might be wrong

**Comment by lasdou** [comment]
Hello, 

On my side I'm using a KUBECONFIG file, in which the `auth-provider` name is set to `gcp` 

And I have this deprecated message too.

I don't know on which auth-provider name I should switch to be able to use the recommended gke-gcloud-auth-plugin. 

any ideas ?

Thank you very much

**Comment by lbanchero** [comment]
Warning started appearing today, followed the instructions and still have the warning. Also in the KUBECONFIG file still have the auth provider set as gcp as @lasdou is mentioning below.

Any ideas?

**Comment by lbanchero** [comment]
I identified the problem, at least for my case... I had a wrong kubectl binary on my machine.

Seems that kubectl command I was using was from docker... I assume minikube.

I had to install kubectl from [this guide](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl) and at the end had this warning: 

> WARNING:   There are other instances of Google Cloud tools on your system PATH.
  Please remove the following to avoid confusion or accidental invocation:
/Applications/Docker.app/Contents/Resources/bin/kubectl

Just deleted that binary and problem solved.

**Comment by groksrc** [comment]
@lbanchero your suggestion worked for me. I had installed minikube with brew and the version of kubectl that came with it was being used by my shell. Here is what I did:

```
$ brew uninstall kubectl --ignore-dependencies
$ gcloud components install kubectl
$ mv /Applications/Docker.app/Contents/Resources/bin/kubectl /Applications/Docker.app/Contents/Resources/bin/kubectl.bak
```

I note that the version installed by gcloud was v1.23.3 but the version I had previous from minikube was 1.24.0

**Comment by mikedd** [comment]
I think the gcloud [documentation](https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke) has the flag set backwards. 

I changed
`USE_GKE_GCLOUD_AUTH_PLUGIN=False` 
to
`USE_GKE_GCLOUD_AUTH_PLUGIN=true`

~/.kube/config looks correct and I don't get the warning.
